### PR TITLE
Transaction pool: Remove futures-diagnose and thread pool

### PR DESCRIPTION
**Pull request by bkchr**
This pr removes `futures-diagnose` as this isn't used anymore. Besides
that the pr also removes the thread pool that was used to validate the
transactions in the background. Instead of this thread pool we now spawn
two separate long running tasks that we use to validate the
transactions. All tasks of the transaction pool are now also spawned as
essential tasks. This means, if any of these tasks is stopping, the node
will stop as well.

polkadot companion: https://github.com/paritytech/polkadot/pull/3204

### CODE REVIEW

File: client/transaction-pool/src/api.rs [hunk 1]
```diff
@@ -65,13 +88,15 @@ impl<Client, Block> FullChainApi<Client, Block> {
 			}
 		});
 
+		let (sender, receiver) = mpsc::channel(1);
```
[the remark this answers is not in the document]
Comment (bkchr) [reply client/transaction-pool/src/api.rs:91]: Yeah, somehow I was dumb. 

0 makes sense

File: client/transaction-pool/src/api.rs [hunk 2]
```diff
@@ -65,13 +88,15 @@ impl<Client, Block> FullChainApi<Client, Block> {
...
+
+		let receiver = Arc::new(Mutex::new(receiver));
+		spawn_validation_pool_task("transaction-pool-task-0", receiver.clone(), spawner);
+		spawn_validation_pool_task("transaction-pool-task-1", receiver, spawner);
```
[the remark this answers is not in the document]
Comment (bkchr) [reply client/transaction-pool/src/api.rs:95]: Initially I thought about having this configurable. However, I also csn not imagine thet the performance is thet bad.

File: client/transaction-pool/src/api.rs (hunk 1 above)
Comment (bkchr) [inline client/transaction-pool/src/api.rs:91]:
```suggestion
		let (sender, receiver) = mpsc::channel(0);
```

**Review by tomusdrw — APPROVED** [review]
lgtm!

**Comment by tomusdrw** [comment]
@tomaka could you take a look if it makes sense to you too?

File: bin/node-template/node/src/service.rs [hunk 3]
```diff
@@ -68,7 +68,7 @@ pub fn new_partial(config: &Configuration) -> Result<sc_service::PartialComponen
 		config.transaction_pool.clone(),
 		config.role.is_authority().into(),
 		config.prometheus_registry(),
-		task_manager.spawn_handle(),
+		task_manager.spawn_essential_handle(),
```
Comment (tomaka) [inline bin/node-template/node/src/service.rs:71]: Makes me wonder if there is any task that is *not* essential.

**Review by tomaka — APPROVED** [review]
:+1:

File: bin/node-template/node/src/service.rs (hunk 3 above)
Comment (bkchr) [reply bin/node-template/node/src/service.rs:71]: I thought the same :D 

But yes, sometimes, short lived stuff is not essential and is fine to end.

**Comment by bkchr** [comment]
bot merge

**Pull request merged**